File: src/main/java/me/miki/shindo/management/mods/impl/AnimationsMod.java
package me.miki.shindo.management.mods.impl;

import lombok.Getter;
import me.miki.shindo.management.language.TranslateText;
import me.miki.shindo.management.settings.config.Property;
import me.miki.shindo.management.settings.config.PropertyType;
import me.miki.shindo.management.mods.Mod;
import me.miki.shindo.management.mods.ModCategory;
import me.miki.shindo.management.settings.impl.BooleanSetting;
import me.miki.shindo.management.settings.impl.NumberSetting;
import me.miki.shindo.management.settings.metadata.SettingRegistry;

public class AnimationsMod extends Mod {

    @Getter
    private static AnimationsMod instance;

    @Property(type = PropertyType.BOOLEAN, translate = TranslateText.BLOCK_HIT)
    private boolean blockHitSetting = true;

    @Property(type = PropertyType.BOOLEAN, translate = TranslateText.PUSHING)
    private boolean pushingSetting = true;

    @Property(type = PropertyType.BOOLEAN, translate = TranslateText.PUSHING_PARTICLES)
    private boolean pushingParticleSetting = true;

    @Property(type = PropertyType.BOOLEAN, translate = TranslateText.SNEAK)
    private boolean sneakSetting = true;

    @Property(type = PropertyType.BOOLEAN, translate = TranslateText.SNEAKSMOOTH)
    private boolean smoothSneakSetting = false;

    @Property(type = PropertyType.NUMBER, translate = TranslateText.SMOOTH_SPEED, min = 0.5, max = 20, step = 0.5)
    private double smoothSneakSpeedSetting = 6;

    @Property(type = PropertyType.BOOLEAN, translate = TranslateText.HEALTH)
    private boolean healthSetting = true;

    @Property(type = PropertyType.BOOLEAN, translate = TranslateText.ARMOR_DAMAGE)
    private boolean armorDamageSetting = false;

    @Property(type = PropertyType.BOOLEAN, translate = TranslateText.ITEM_SWITCH)
    private boolean itemSwitchSetting = false;

    @Property(type = PropertyType.BOOLEAN, translate = TranslateText.ROD)
    private boolean rodSetting = true;

    public AnimationsMod() {
        super(TranslateText.OLD_ANIMATION, TranslateText.OLD_ANIMATION_DESCRIPTION, ModCategory.RENDER, "oldoam1.7smoothsneak");

        instance = this;
    }

    public float getSmoothSneakSpeedSetting() {
        return (float) smoothSneakSpeedSetting;
    }

    public BooleanSetting getBlockHitSetting() {
        return SettingRegistry.getBooleanSetting(this, "blockHitSetting");
    }

    public BooleanSetting getPushingSetting() {
        return SettingRegistry.getBooleanSetting(this, "pushingSetting");
    }

    public BooleanSetting getPushingParticleSetting() {
        return SettingRegistry.getBooleanSetting(this, "pushingParticleSetting");
    }

    public BooleanSetting getSneakSetting() {
        return SettingRegistry.getBooleanSetting(this, "sneakSetting");
    }

    public BooleanSetting getSmoothSneakSetting() {
        return SettingRegistry.getBooleanSetting(this, "smoothSneakSetting");
    }

    public NumberSetting getSmoothSneakSpeedSettingProperty() {
        return SettingRegistry.getNumberSetting(this, "smoothSneakSpeedSetting");
    }

    public BooleanSetting getHealthSetting() {
        return SettingRegistry.getBooleanSetting(this, "healthSetting");
    }

    public BooleanSetting getArmorDamageSetting() {
        return SettingRegistry.getBooleanSetting(this, "armorDamageSetting");
    }

    public BooleanSetting getItemSwitchSetting() {
        return SettingRegistry.getBooleanSetting(this, "itemSwitchSetting");
    }

    public BooleanSetting getRodSetting() {
        return SettingRegistry.getBooleanSetting(this, "rodSetting");
    }
}

File: src/main/kotlin/me/miki/shindo/management/mods/impl/AnimationsMod.kt
package me.miki.shindo.management.mods.impl

import me.miki.shindo.management.language.TranslateText
import me.miki.shindo.management.mods.Mod
import me.miki.shindo.management.mods.ModCategory
import me.miki.shindo.management.nanovg.font.Shinconic
import me.miki.shindo.management.settings.config.Property
import me.miki.shindo.management.settings.config.PropertyType
import me.miki.shindo.management.settings.impl.BooleanSetting
import me.miki.shindo.management.settings.impl.NumberSetting
import me.miki.shindo.management.settings.metadata.SettingRegistry.getBooleanSetting
import me.miki.shindo.management.settings.metadata.SettingRegistry.getNumberSetting

class AnimationsMod :
    Mod(
        TranslateText.OLD_ANIMATION,
        TranslateText.OLD_ANIMATION_DESCRIPTION,
        ModCategory.RENDER,
        Shinconic.MOD_ANIMATIONS,
        "oldoam1.7smoothsneak",
    ) {
    @Property(type = PropertyType.BOOLEAN, translate = TranslateText.BLOCK_HIT)
    @JvmField
    var blockHitSetting = false

    @Property(type = PropertyType.BOOLEAN, translate = TranslateText.PUSHING)
    @JvmField
    var pushingSetting = false

    @Property(type = PropertyType.BOOLEAN, translate = TranslateText.PUSHING_PARTICLES)
    @JvmField
    var pushingParticleSetting = false

    @Property(type = PropertyType.BOOLEAN, translate = TranslateText.SNEAK)
    @JvmField
    var sneakSetting = false

    @Property(type = PropertyType.BOOLEAN, translate = TranslateText.SNEAKSMOOTH)
    @JvmField
    var smoothSneakSetting = false

    @Property(type = PropertyType.NUMBER, translate = TranslateText.SMOOTH_SPEED, min = 0.5, max = 20.0, step = 0.5)
    private val smoothSneakSpeedSetting = 6.0

    @Property(type = PropertyType.BOOLEAN, translate = TranslateText.HEALTH)
    @JvmField
    var healthSetting = false

    @Property(type = PropertyType.BOOLEAN, translate = TranslateText.ARMOR_DAMAGE)
    @JvmField
    var armorDamageSetting = false

    @Property(type = PropertyType.BOOLEAN, translate = TranslateText.ITEM_SWITCH)
    @JvmField
    var itemSwitchSetting = false

    @Property(type = PropertyType.BOOLEAN, translate = TranslateText.ROD)
    @JvmField
    var rodSetting = false

    init {
        instance = this
    }

    fun getSmoothSneakSpeedSetting(): Float = smoothSneakSpeedSetting.toFloat()

    val smoothSneakSpeedSettingProperty: NumberSetting?
        get() = getNumberSetting(this, "smoothSneakSpeedSetting")

    companion object {
        @JvmField
        var instance: AnimationsMod? = null
    }

    fun getBlockHitSetting(): BooleanSetting? = getBooleanSetting(this, "blockHitSetting")

    fun getPushingSetting(): BooleanSetting? = getBooleanSetting(this, "pushingSetting")

    fun getPushingParticleSetting(): BooleanSetting? = getBooleanSetting(this, "pushingParticleSetting")

    fun getSneakSetting(): BooleanSetting? = getBooleanSetting(this, "sneakSetting")

    fun getSmoothSneakSetting(): BooleanSetting? = getBooleanSetting(this, "smoothSneakSetting")

    fun getHealthSetting(): BooleanSetting? = getBooleanSetting(this, "healthSetting")

    fun getArmorDamageSetting(): BooleanSetting? = getBooleanSetting(this, "armorDamageSetting")

    fun getItemSwitchSetting(): BooleanSetting? = getBooleanSetting(this, "itemSwitchSetting")

    fun getRodSetting(): BooleanSetting? = getBooleanSetting(this, "rodSetting")
}
